crates/core/src/models/rule.rs

```diff
@@ -307,15 +314,23 @@ impl InstantiatedRule {
   /// Check if this rule applies to the given file path based on its directory scope
   pub(crate) fn applies_to_file(&self, file_path: &std::path::Path) -> bool {
     match &self.directory_scope {
-      None => true, // Global scope - applies to all files
-      Some(scope_dir) => {
-        // Directory scope - check if file is in the exact same directory (non-recursive)
+      DirectoryScope::Global => true, // Global scope - applies to all files
+      DirectoryScope::Directory(scope_dir) => {
+        // Directory scope - check exact same directory only
         if let Some(file_dir) = file_path.parent() {
           file_dir == scope_dir
         } else {
           false
         }
       }
+      DirectoryScope::DirectoryRecursive(scope_dir) => {
+        // DirectoryRecursive scope - check same directory or subdirectories
+        if let Some(file_dir) = file_path.parent() {
+          file_dir == scope_dir || file_dir.starts_with(scope_dir)
```

Comment:
should we use a Path-like abstraction to check that the sub directories have `scope_dir` as parent? `starts_with()` seems a bit fragile and we can certainly come up with some counter examples where it would fail.